passes/hierarchy/hierarchy.cc

```diff
@@ -729,14 +736,24 @@ struct HierarchyPass : public Pass {
 				auto_top_mode = true;
 				continue;
 			}
+			if (args[argidx] == "-chparam"  && argidx+2 < args.size()) {
+				const std::string &key = args[++argidx];
+				const std::string &value = args[++argidx];
+				auto r = parameters.emplace(key, value);
+				if (!r.second) {
+					log_warning_noprefix("-chparam %s already specified: overwriting.\n", key.c_str());
```

Comment:
That should probably simply be `log_warning()`.

Reply:
Good spot!